packages/grafana-schema/src/schema/dashboard/dashboard.gen.ts

```diff
@@ -0,0 +1,216 @@
+// This file is autogenerated. DO NOT EDIT.
+//
+// Derived from the Thema lineage at pkg/coremodel/dashboard/lineage.cue
+
+export interface Dashboard {
+  annotations?: {
+    list: AnnotationQuery[];
+  };
+  description?: string;
+  editable: boolean;
+  fiscalYearStartMonth?: number;
+  gnetId?: string;
+  graphTooltip: DashboardCursorSync;
+  id?: number;
+  links?: DashboardLink[];
+  liveNow?: boolean;
+  panels?: Panel | {
+      type: 'graph';
+    } | {
+      type: 'heatmap';
+    } | {
+      type: 'row';
+      collapsed: boolean;
+      id: number;
+      panels: Panel | {
+          type: 'graph';
+        } | {
+          type: 'heatmap';
+        }[];
+    }[];
+  refresh?: string | false;
+  schemaVersion: number;
+  style: 'light' | 'dark';
+  tags?: string[];
+  templating?: {
+    list: VariableModel[];
+  };
+  time?: {
+    from: string;
+    to: string;
+  };
+  timepicker?: {
+    collapse: boolean;
+    enable: boolean;
+    hidden: boolean;
+    refresh_intervals: string[];
+  };
+  timezone?: 'browser' | 'utc' | '';
+  title?: string;
+  uid?: string;
+  version?: number;
+  weekStart?: string;
+}
+
+export const defaultDashboard: Partial<Dashboard> = {
+  editable: true,
+  graphTooltip: DashboardCursorSync.Off,
+  links: [],
+  panels: [],
+  schemaVersion: 36,
+  style: 'dark',
+  tags: [],
+  timezone: 'browser',
+};
+
+export interface AnnotationQuery {
+  builtIn: number;
+  datasource: {};
+  enable: boolean;
+  hide?: boolean;
+  iconColor?: string;
+  name?: string;
+  rawQuery?: string;
+  showIn: number;
+  target?: {};
+  type: string;
+}
+
+export const defaultAnnotationQuery: Partial<AnnotationQuery> = {
+  builtIn: 0,
+  enable: true,
+  hide: false,
+  showIn: 0,
+  type: 'dashboard',
+};
+
+export interface VariableModel {
+  label?: string;
+  name: string;
+  type: VariableType;
+}
+
+export interface DashboardLink {
+  asDropdown: boolean;
+  icon?: string;
+  includeVars: boolean;
+  keepTime: boolean;
+  tags: string[];
+  targetBlank: boolean;
+  title: string;
+  tooltip?: string;
+  type: DashboardLinkType;
+  url?: string;
+}
+
+export const defaultDashboardLink: Partial<DashboardLink> = {
+  asDropdown: false,
+  includeVars: false,
+  keepTime: false,
+  tags: [],
+  targetBlank: false,
+};
+
+export type DashboardLinkType = 'link' | 'dashboards';
+
+export type VariableType = 'query' | 'adhoc' | 'constant' | 'datasource' | 'interval' | 'textbox' | 'custom' | 'system';
+
+export enum FieldColorModeId {
+  Continuous-GrYlRd = 'continuous-GrYlRd',
+  Fixed = 'fixed',
+  Palette-Classic = 'palette-classic',
+  Palette-Saturated = 'palette-saturated',
+  Thresholds = 'thresholds',
+}
+
+export type FieldColorSeriesByMode = 'min' | 'max' | 'last';
+
+export interface FieldColor {
```

Comment:
How can we give all these a different home?  dashboard.gen.ts.... not quite appropriate

All the field/color/etc stuff can live outside the dashboard/panel context

Reply:
It's the same basic answer as with panel plugins - move them into some reusable, external set of definitions, such as those in the parent directory (`grafana-schema/src/schema`), and import+reference those from within the dashboard lineage.

That said, referencing anything outside of the lineage is scary, even (and perhaps especially) in the plugin case. The deep work required to separate out and formalize Thema suggested a couple possible solutions to me, but also made me more concerned about the negative impact of allowing direct external references. As long as it's an ergonomics and not a functionality issue, it would be better to keep all the declarations clustered together within the dashboard lineage. Once i get through generic composition (grafana/thema#8), which is now quite near the top of my priority list, i should have a much clearer sense of how to tackle this.

Reply:
Though...there _may_ be a relatively easy short-term solution. Because we generate the sub-members in a separate cuetsy call (`GenerateAST()`) from the one that generates the dashboard interface itself (`GenerateSingleAST()`), it _might_ be straightforward to just write the other members out to a separate file in the parent dir, and then just insert import statements into the `dashboard.gen.ts`.

Because we're only having the latest schema version in the TS anyway, this doesn't really hurt anything. And this keeps all the declarations together within the Thema lineage. Might satisfy both goals.